rules/Renaming/Rector/Name/RenameClassRector.php

```diff
@@ -109,4 +111,58 @@ public function configure(array $configuration): void
 
         $this->renamedClassesDataCollector->addOldToNewClasses($configuration);
     }
+
+    /**
+     * @param Node[] $nodes
+     * @return null|Node[]
+     */
+    public function afterTraverse(array $nodes): ?array
+    {
+        foreach ($nodes as $node) {
+            if ($node instanceof Namespace_) {
+                return parent::afterTraverse($nodes);
+            }
+
+            if ($node instanceof FileWithoutNamespace) {
+                foreach ($node->stmts as $stmt) {
+                    if ($stmt instanceof Namespace_) {
+                        $this->restructureUnderNamespace($node);
+                        return $node->stmts;
+                    }
+                }
+
+                return parent::afterTraverse($nodes);
+            }
+        }
+
+        return parent::afterTraverse($nodes);
+    }
+
+    private function restructureUnderNamespace(FileWithoutNamespace $fileWithoutNamespace): void
+    {
+        $stmts = array_reverse($fileWithoutNamespace->stmts, true);
+        $isBeforeNamespace = false;
+        $stmtsBeforeNamespace = [];
+        $namepace = null;
+
+        foreach ($stmts as $key => $stmt) {
+            if ($stmt instanceof Namespace_) {
+                $isBeforeNamespace = true;
+                $namepace = $stmt;
+                continue;
+            }
+
+            if ($isBeforeNamespace && ! $stmt instanceof Declare_) {
+                $stmtsBeforeNamespace[] = $stmt;
+                unset($stmts[$key]);
+            }
+        }
+
+        if ($stmtsBeforeNamespace === [] || ! $namepace instanceof Namespace_) {
+            return;
+        }
+
+        $namepace->stmts = array_values([...$stmtsBeforeNamespace, ...$namepace->stmts]);
+        $fileWithoutNamespace->stmts = array_values(array_reverse($stmts, true));
```

Comment:
The `afterTraverse()` on `RenameClassRector` to restructure under namespace is the strict 👍